lib/matplotlib/patches.py

```diff
@@ -1659,14 +1661,28 @@ def theta_stretch(theta, scale):
             theta = np.deg2rad(theta)
             x = np.cos(theta)
             y = np.sin(theta)
-            return np.rad2deg(np.arctan2(scale * y, x))
-        theta1 = theta_stretch(self.theta1, width / height)
-        theta2 = theta_stretch(self.theta2, width / height)
+            stheta = np.rad2deg(np.arctan2(scale * y, x))
+            # arctan2 has the range [-pi, pi], we expect [0, 2*pi]
+            return (stheta + 360) % 360
+
+        theta1 = self.theta1
+        theta2 = self.theta2
+
+        if (
+            # if we need to stretch the angles because we are distorted
+            width != height
+            # and we are not doing a full circle
+            and not (theta1 != theta2 and theta1 % 360 == theta2 % 360)
+        ):
+            theta1 = theta_stretch(self.theta1, width / height)
+            theta2 = theta_stretch(self.theta2, width / height)
 
         # Get width and height in pixels
-        width, height = self.get_transform().transform((width, height))
+        pwidth, pheight = (self.get_transform().transform((width, height)) -
+                           self.get_transform().transform((0, 0)))
```

Comment:
Because of 
https://github.com/matplotlib/matplotlib/blob/c3b37caae291121aed9e53535a686c430ca2aed8/lib/matplotlib/patches.py#L1398-L1414
I think this maybe should be:
```suggestion
        pwidth, pheight = (self.get_transform().transform((1, 1)) -
                           self.get_transform().transform((-1, -1)))
```
but it may not correctly handle angle.

Reply:
Solved this a different way (used just the data -> screen part of the transform).